tests/ryzenai/test_modeling.py

```diff
@@ -67,222 +83,130 @@ def test_save_model(self):
             model.save_pretrained(tmpdirname)
             folder_contents = os.listdir(tmpdirname)
             self.assertTrue("ResNet_int.onnx" in folder_contents)
 
 
-class RyzenAIModelForImageClassificationIntegrationTest(unittest.TestCase):
+class RyzenAIModelForImageClassificationIntegrationTest(unittest.TestCase, RyzenAITestCaseMixin):
     @parameterized.expand(RYZEN_PREQUANTIZED_MODEL_IMAGE_CLASSIFICATION)
     @pytest.mark.prequantized_model_test
     def test_model(self, model_id):
-        set_seed(SEED)
+        cache_dir = DEFAULT_CACHE_DIR
+        cache_key = model_id.replace("/", "_").lower()
 
-        all_files = huggingface_hub.list_repo_files(model_id, repo_type="model")
+        file_name, ort_input, input_name = load_model_and_input(model_id)
 
-        file_name = [name for name in all_files if name.endswith(".onnx")][0]
+        vaip_config = DEFAULT_VAIP_CONFIG
+        outputs_ipu, outputs_cpu = self.prepare_outputs(
+            model_id, RyzenAIModelForImageClassification, ort_input, vaip_config, cache_dir, cache_key, file_name
+        )
 
-        onnx_model_path = huggingface_hub.hf_hub_download(model_id, file_name)
-        model = onnx.load(onnx_model_path)
+        self.assertIn("logits", outputs_ipu)
+        self.assertIn("logits", outputs_cpu)
 
-        input_shape = model.graph.input[0].type.tensor_type.shape.dim
-        input_shape = [dim.dim_value for dim in input_shape]
-        input_shape[0] = 1
-
-        ort_input = DummyInputGenerator.random_float_tensor(input_shape, framework="np")
-
-        def run(model_id, file_name, ort_input, use_cpu_runner, compile_reserve_const_data):
-            os.environ["XLNX_ENABLE_CACHE"] = "0"
-            os.environ["XLNX_USE_SHARED_CONTEXT"] = "1"
+        self.assertTrue(np.allclose(outputs_ipu.logits, outputs_cpu.logits, atol=1e-4))
 
-            os.environ["USE_CPU_RUNNER"] = "1" if use_cpu_runner else "0"
-            os.environ["VAIP_COMPILE_RESERVE_CONST_DATA"] = "1" if compile_reserve_const_data else "0"
-
-            model = RyzenAIModelForImageClassification.from_pretrained(
-                model_id, file_name=file_name, vaip_config=".\\tests\\ryzenai\\vaip_config.json"
-            )
-
-            outputs = model(ort_input)
-
-            return outputs
-
-        output_ipu = run(model_id, file_name, ort_input, use_cpu_runner=0, compile_reserve_const_data=0)
-        output_cpu = run(model_id, file_name, ort_input, use_cpu_runner=1, compile_reserve_const_data=1)
-
-        self.assertIn("logits", output_ipu)
-        self.assertIn("logits", output_cpu)
-
-        self.assertTrue(np.allclose(output_ipu.logits, output_cpu.logits, atol=1e-4))
+        current_ops = self.get_ops(cache_dir, cache_key)
+        baseline_ops = self.get_baseline_ops(cache_key)
+        self.assertEqual(baseline_ops["dpu"], current_ops["dpu"], "DPU operators do not match!")
```

Comment:
maybe rather `baseline_ops["dpu"] <= current_ops["dpu"]`?

Reply:
This `current_ops` should be same as baseline until version change so it would be better check for equality. This would also ensure to update in case SDK version is changed.